wcfsetup/install/files/lib/acp/form/CaptchaQuestionAddForm.class.php

```diff
@@ -3,198 +3,104 @@
-                'answers' => I18nHandler::getInstance()->isPlainValue('answers') ? I18nHandler::getInstance()->getValue('answers') : '',
-                'isDisabled' => $this->isDisabled,
-                'question' => I18nHandler::getInstance()->isPlainValue('question') ? I18nHandler::getInstance()->getValue('question') : '',
-            ]),
-        ]);
-        $returnValues = $this->objectAction->executeAction();
-        $questionID = $returnValues['returnValues']->questionID;
-
-        // set i18n values
-        $questionUpdates = [];
-        if (!I18nHandler::getInstance()->isPlainValue('question')) {
-            I18nHandler::getInstance()->save(
-                'question',
-                'wcf.captcha.question.question.question' . $questionID,
-                'wcf.captcha.question',
-                1
-            );
-
-            $questionUpdates['question'] = 'wcf.captcha.question.question.question' . $questionID;
-        }
-        if (!I18nHandler::getInstance()->isPlainValue('answers')) {
-            I18nHandler::getInstance()->save(
-                'answers',
-                'wcf.captcha.question.answers.question' . $questionID,
-                'wcf.captcha.question',
-                1
-            );
-
-            $questionUpdates['answers'] = 'wcf.captcha.question.answers.question' . $questionID;
-        }
-
-        if (!empty($questionUpdates)) {
-            $questionEditor = new CaptchaQuestionEditor($returnValues['returnValues']);
-            $questionEditor->update($questionUpdates);
-        }
-
-        $this->saved();
-
-        // reset values
-        I18nHandler::getInstance()->reset();
-        $this->isDisabled = 0;
-
-        // show success message
-        WCF::getTPL()->assign([
-            'success' => true,
-            'objectEditLink' => LinkHandler::getInstance()->getControllerLink(
-                CaptchaQuestionEditForm::class,
-                ['id' => $questionID]
-            ),
+        parent::createForm();
+
+        $this->form->appendChildren([
+            FormContainer::create('data')
+                ->appendChildren([
+                    TextFormField::create('question')
+                        ->label('wcf.acp.captcha.question.question')
+                        ->maximumLength(255)
+                        ->required()
+                        ->i18n()
+                        ->languageItemPattern('wcf.captcha.question.question.question\d+'),
+                    MultilineTextFormField::create('answers')
+                        ->label('wcf.acp.captcha.question.answers')
+                        ->required()
+                        ->i18n()
+                        ->languageItemPattern('wcf.captcha.question.answers.question\d+')
+                        ->addValidator(new FormFieldValidator('regex', function (MultilineTextFormField $formField) {
+                            if (!empty($formField->getValidationErrors())) {
```

Comment:
`empty()`

Reply:
Why? It is meant to validate in case the default validation has passed, which means it should do a early return in case there are validation errors. 🤔

Reply:
Sorry, I was unclear here. My comment was meant to say “no empty please”, implicitly referencing the other review:

https://github.com/WoltLab/WCF/pull/5252#discussion_r1090336095